designer/server/src/main/scala/pl/touk/nussknacker/ui/process/newdeployment/DeploymentServiceNG.scala

```diff
@@ -0,0 +1,114 @@
+package pl.touk.nussknacker.ui.process.newdeployment
+
+import cats.data.EitherT
+import db.util.DBIOActionInstances._
+import pl.touk.nussknacker.engine.api.deployment.DataFreshnessPolicy
+import pl.touk.nussknacker.engine.api.deployment.StateStatus.StatusName
+import pl.touk.nussknacker.engine.api.process.ProcessIdWithName
+import pl.touk.nussknacker.engine.deployment.ExternalDeploymentId
+import pl.touk.nussknacker.security.Permission
+import pl.touk.nussknacker.ui.db.entity.ProcessEntityData
+import pl.touk.nussknacker.ui.error.{
+  CommentValidationErrorNG,
+  GetDeploymentStatusError,
+  NoPermissionError,
+  RunDeploymentError
+}
+import pl.touk.nussknacker.ui.process.deployment.{
+  DeploymentCommand,
+  DeploymentService,
+  RunDeploymentCommand,
+  RunDeploymentCommandNG
+}
+import pl.touk.nussknacker.ui.process.newdeployment.DeploymentEntityFactory.DeploymentEntityData
+import pl.touk.nussknacker.ui.process.repository.{CommentValidationError, DBIOActionRunner, ScenarioRepository}
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import java.sql.Timestamp
+import java.time.Clock
+import scala.concurrent.{ExecutionContext, Future}
+
+// TODO: we should replace DeploymentService by this class after we split Deployments and Activities
+class DeploymentServiceNG(
+    scenariosRepository: ScenarioRepository,
+    deploymentRepository: DeploymentRepository,
+    // TODO: we shouldn't call legacy service, instead we should new service from the legacy one
+    legacyDeploymentService: DeploymentService,
+    dbioRunner: DBIOActionRunner,
+    clock: Clock
+)(implicit ec: ExecutionContext) {
+
+  def processCommand(
+      command: DeploymentCommand
+  ): EitherT[Future, RunDeploymentError, Future[Option[ExternalDeploymentId]]] = {
+    command match {
+      case command: RunDeploymentCommandNG =>
+        runDeployment(command)
+    }
+  }
+
+  private def runDeployment(
+      command: RunDeploymentCommandNG
+  ): EitherT[Future, RunDeploymentError, Future[Option[ExternalDeploymentId]]] = {
+    dbioRunner.runInTransactionE(
+      for {
+        scenarioMetadata <- scenariosRepository.getScenarioMetadata(command.scenarioName)
+        _ <- EitherT.fromEither(
+          Either.cond(command.user.can(scenarioMetadata.processCategory, Permission.Deploy), (), NoPermissionError)
+        )
+        _ <- deploymentRepository.saveDeployment(
+          DeploymentEntityData(command.id, scenarioMetadata.id, Timestamp.from(clock.instant()), command.user.id)
```

Comment:
I'm not sure about `command.user.id`. In other tables we use `command.user.username` and I'm pretty sure that it is bad. `username` causes that when some user changes the displayable name (e.g. surname part), in historical data there will be the outdated name. But on the other hand it is hard to predict when w fix that